odoo/tests/common.py

```diff
@@ -311,6 +311,14 @@ class BaseCase(TreeCase, MetaCase('DummyCase', (object,), {})):
     longMessage = True      # more verbose error message by default: https://www.odoo.com/r/Vmh
     warm = True             # False during warm-up phase (see :func:`warmup`)
 
+    def setUp(self):
+        from odoo.addons.iap.tools import iap_tools
+
+        def iap_jsonrpc(url, method='call', params=None, timeout=15):
+            raise exceptions.AccessError("Unavailable during tests.")
+
+        iap_tools.iap_jsonrpc = Mock(side_effect=iap_jsonrpc)
+
```

Comment:
Not need to import iap_tools and monkey patch it by hand.
Also, your monkey patch is not removed after the test.You can remove it with the `addCleanup` method like:
```
iap_patch = patch('odoo.addons.iap.tools.iap_tools.iap_jsonrpc', iap_jsonrpc)
iap_patch.start()
self.addCleanup(iap_patch.stop)

Reply:
Thank you, I updated the code with your suggestion.